dialogue-apache-hc4-client/src/test/java/com/palantir/dialogue/hc4/ApacheHttpClientChannelsTest.java

```diff
@@ -58,6 +59,26 @@ public void close_works() throws Exception {
         assertThatThrownBy(() -> again.get()).hasMessageContaining("Connection pool shut down");
     }
 
+    @Test
+    public void metrics() throws Exception {
+        ClientConfiguration conf = TestConfigurations.create("http://unused");
+
+        try (ApacheHttpClientChannels.CloseableClient client =
+                ApacheHttpClientChannels.createCloseableHttpClient(conf)) {
+
+            Channel channel = ApacheHttpClientChannels.createSingleUri("http://neverssl.com", client);
+            ListenableFuture<Response> response =
+                    channel.execute(new TestEndpoint(), Request.builder().build());
+            assertThat(Futures.getUnchecked(response).code()).isEqualTo(200);
+
+            assertThat(client.getNumRoutes()).describedAs("routes").isEqualTo(1);
+            assertThat(client.getPoolStats().getAvailable())
+                    .describedAs("available")
+                    .isEqualTo(0);
```

Comment:
I think graphing the number `available` will be kinda interesting, as this should tell us how quickly we could handle bursts of traffic 

```java
 /**
     * Gets the number idle persistent connections.
     * <p>
     * The total number of connections in the pool is equal to {@code available} plus {@code leased}.
     * </p>
     *
     * @return number idle persistent connections.
     */
    public int getAvailable() {
```